src/api/objects/ObjectAPI.js

```diff
@@ -638,19 +638,37 @@ export default class ObjectAPI {
         return false;
     }
 
-    getOriginalPath(identifier, path = []) {
-        return this.get(identifier).then((domainObject) => {
-            path.push(domainObject);
-            let location = domainObject.location;
+    #pathContainsDomainObject(keyStringToCheck, path) {
+        if (!keyStringToCheck) {
+            return false;
+        }
 
-            if (location) {
-                return this.getOriginalPath(utils.parseKeyString(location), path);
-            } else {
-                return path;
-            }
+        return path.some(pathElement => {
+            const identifierToCheck = utils.parseKeyString(keyStringToCheck);
+
+            return this.areIdsEqual(identifierToCheck, pathElement.identifier);
         });
     }
 
+    /**
+     * Given an identifier, constructs the original path by walking up its parents
+     * @param {module:openmct.ObjectAPI~Identifier} identifier
+     * @param {Array<module:openmct.DomainObject>} path an array of path objects
```

Comment:
This is actually an array of domain objects right?

This is an interesting method, because although it's called `getOriginalPath`, it's actually returning a list of all ancestors to the given domainObject? I'm wondering if there is any value in some kind of `Path` object that can provide this information as either a list of ancestors, or a navigable URL, or otherwise... just thinking out loud

Reply:
> I'm wondering if there is any value in some kind of Path object that can provide this information as either a list of ancestors, or a navigable URL, or otherwise

It does seem to be a pretty common pattern throughout the OpenMCT framework to transform these path arrays to URLs and the like. Having object methods to deal with this would be handy!